### ISSUE TITLE
Fix problem when website have a top level domain of two words

### ISSUE THREAD

**Issue by heliomar-pena** [heliomar-pena/nsfw-proxy#3]
# Overview

The current logic takes the two last words of the host as if it was the second level domain and the top level domain.

> [!NOTE]
> For example `https://de.google.com` is converted to `google.com`

But this cause problems if the top level domain includes more than one word, for example, if the site was `www.google.co.uk` we would get only `.co.uk`, and we couldn't compare it with our blacklisted sites.

We need to found a better way to extract this information properly that fit better with all the types of domains that could exists.

## Extra details

Some domains I found are:

```
.co.uk
.com.au
.co.nz
.com.br
```

There is also a problem with those websites that creates spaces as subdomains, in that cases the subdomain could be important, for example: "blogspot"

`nsfwsite.blogspot.com`

## References

- [Top level domain lists](https://tld-list.com/tlds-from-a-z)

**Comment by heliomar-pena** [comment]
Here is a suggestion from reddit's user: 

```python
blacklisted = any(map(lambda x: host == x or host.endswith("." + x), blacklist))
```

and

```python
blacklisted = False
domain_parts = domain.split(".")
for i in range(len(domain_parts)):
    if ".".join(domain_parts[-i:]) in blacklist:
        blacklisted = True
        break
```

Full comment:

![image](https://github.com/user-attachments/assets/c7f82faf-3092-4bda-ab65-eb639f6fe01c)

**Comment by heliomar-pena** [comment]
I did take a look to this problem, the proposed solution is too slow for sites that have a lot of images as pinterest of reddit. And the solution that was implemented before have the problems mentioned on the task

So I decided remove the support to undefined subdomains until there was a optimization for these cases as cache or similar. Since determining if `es.google.com`, is the same that `google.com` could be resource expensive for the user's computer.

Instead, we can define the pages subdomains that are NSFW explicitly on the blacklist. This will make the match fastest than before.
